Plugins/Object/Object.cpp

```diff
@@ -1003,4 +1005,29 @@ ArgumentStack Object::GetCurrentAnimation(ArgumentStack&& args)
     return Services::Events::Arguments(retVal);
 }
 
+ArgumentStack Object::GetAILevel(ArgumentStack&& args)
+{
+    int32_t retVal = -1;
+
+    if (auto *pObject = object(args))
+    {
+        retVal = pObject->m_nAILevel;
+    }
+
+    return Services::Events::Arguments(retVal);
+}
+
+ArgumentStack Object::SetAILevel(ArgumentStack&& args)
+{
+    if (auto *pObject = object(args))
+    {
+        const auto nLevel = Services::Events::ExtractArgument<int32_t>(args);
+        ASSERT_OR_THROW(nLevel >= -1);
+        ASSERT_OR_THROW(nLevel <= 4);
+        pObject->m_nAILevel = nLevel;
```

Comment:
You (also?) want to call the method in the AIMaster, otherwise this won't actually do much:
```c

    CServerAIMaster* ai = Globals::AppManager()->m_pServerExoApp->GetServerAIMaster();
    ai->SetAILevel(pObject, nLevel);
```

Reply:
Thank you. Updated. The setting of `pObject->m_nAILevel` directly does not seem necessary when calling SetAILevel.